src/api/eidas/controller.ts

```diff
@@ -85,27 +84,26 @@ export default class Controller {
   }
 
   static async putEidasKeys(
-    did: string,
-    opts: EidasKeysData
+    eidasQecId: string,
+    opts: EidasKeysInput
   ): Promise<RedisInsertion> {
-    if (
-      !opts ||
-      !opts.p12 ||
-      !opts.keyType ||
-      opts.keyType !== constants.KeyTypes.RSA
-    )
+    if (!opts || !opts.did || !opts.eidasQec)
       throw new BadRequestError(BadRequestError.defaultTitle, {
         detail: ApiErrorMessages.BAD_INPUT_EIDAS_KEYS_PARAMS,
       });
-    if (!did)
+    if (!eidasQecId)
       throw new BadRequestError(BadRequestError.defaultTitle, {
         detail: ApiErrorMessages.MISSING_PUT_ID_PARAMS,
       });
-    const previousKeys = await redis.get(did);
-    await redis.set(did, JSON.stringify(opts));
-    const returnOpts = { ...opts, did };
+    const previousKeys = await redis.get(opts.did);
+    const eidasKeysData: EidasKeysData = {
+      ...opts,
+      eidasQecId,
+    };
+    await redis.set(opts.did, JSON.stringify(eidasKeysData));
+
     return {
-      eidasKeysData: returnOpts,
+      id: opts.did,
```

Comment:
Shouldn't we return the identifier of the key instead of the did associated to the key?

Reply:
That was what we had before, and I agree that should be the data to be returned. However, the current Open API specs agreed on the eidas-interop group, shows the did as the returning data...

Reply:
Ok. We'll leave it as agreed on specs then. Just wanted to be sure.